packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -369,11 +374,21 @@ defineProperties(Element.prototype, {
             const nodeList = arrayFromCollection(
                 elementQuerySelectorAll.apply(this, ArraySlice.call(arguments) as [string])
             );
-            const filteredResults = getFilteredNodeListQueryResult(
-                this,
-                nodeList,
-                ENABLE_NODE_LIST_PATCH
-            );
+            let filteredResults;
+
+            if (featureFlags.ENABLE_NODE_LIST_PATCH) {
```

Comment:
probably the if should set a let binding, and then below the else, you call that api, otherwise the code duplication is a lot more... and hard to keep it in sync

Reply:
done